node/service-name-handler.js

```diff
@@ -42,6 +42,49 @@ function TChannelServiceNameHandler(options) {
 
 TChannelServiceNameHandler.prototype.type = 'tchannel.service-name-handler';
 
+TChannelServiceNameHandler.prototype.handleLazily = function handleLazily(conn, reqFrame) {
+    var self = this;
+
+    var res = reqFrame.bodyRW.lazy.readService(reqFrame);
+    if (res.err) {
+        // TODO: stat?
+        self.channel.logger.warn('failed to lazy read frame serviceName', conn.extendLogInfo({
+            error: res.err
+        }));
+        // TODO: protocol error instead?
+        self._sendLazyErrorFrame(conn, reqFrame, 'BadRequest', 'failed to read serviceName');
+        return false;
+    }
+
+    var serviceName = res.value;
+    if (!serviceName) {
+        self.channel.logger.warn('missing service name in lazy frame', conn.extendLogInfo({}));
```

Comment:
lets try and pull out callerName & arg1.

Can we implement `reqFrame.extendLogInfo()` ?